shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -140,7 +140,8 @@
   NSObject<FlutterPlatformViewFactory>* factory = factories_[viewType].get();
   if (factory == nil) {
     result([FlutterError errorWithCode:@"unregistered_view_type"
-                               message:@"trying to create a view with an unregistered type"
+                               message:@"trying to create a view with an unregistered type. Have "
+                                       @"you called GeneratedPluginRegistrant.register?"
```

Comment:
@cyanglaz in the wild we saw this error fixed by calling `GeneratedPluginRegistrant.register` https://github.com/flutter/flutter/issues/100241#issuecomment-1074686903 but perhaps there's also some `-registerViewFactory:` to be done in the darwin case?

Reply:
It is already available: https://github.com/flutter/engine/blob/main/shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h#L309

Reply:
Sorry, I know it exists, the question is when this error happens is it reasonable to say "you need to call `GeneratedPluginRegistrant.register`" because it's missing (add-to-app) or do they also need to call `-registerViewFactory:` somewhere? We want this suggestion to actually be actionable.

Reply:
Ah ok.

The `registerViewFactory:` is only meant to be called by the author of the PlatformView. If the user only uses a PlatformView plugin, they wouldn't be able to call `registerViewFactory:` as they don't own the factory, so showing this to the app developer who uses the PlatformView plugin might cause confusion.
I think only include `GeneratedPluginRegistrant.register` is fine for app developers.

We might be able to add some checks for the plugin/PlatformView authors to remind them to call `registerViewFactory` in `registerWithRegistrar`. 

In addition to that, I think this part is also confusing: "trying to create a view with an unregistered type".
What view?(A view means lots of things, especially in add-to-app) Who or what tried to create a view? Maybe something like:

```
A UIKitView widget is trying to create a PlatformView with an unregistered type: <some type>. 
If you are the author of the PlatformView, make sure `registerViewFactory` is invoked. See: https://docs.flutter.dev/development/platform-integration/platform-views?tab=ios-platform-views-objective-c-tab#on-the-platform-side-1 for more details.
If you are not the author of the PlatformView, make sure to call `GeneratedPluginRegistrant.register`. 

```